tools/generators/rc-caret/index.ts

```diff
@@ -0,0 +1,106 @@
+import { Tree, updateJson, getProjects, formatFiles } from '@nrwl/devkit';
+import * as semver from 'semver';
+import { VersionBumpGeneratorSchema } from './schema';
+import { getProjectConfig, isPackageVersionConverged, printUserLogs, UserLog } from '../../utils';
+import { PackageJson } from '../../types';
+
+export default async function (host: Tree, schema: VersionBumpGeneratorSchema) {
+  const userLog: UserLog = [];
+  const validatedSchema = validateSchema(host, schema);
+
+  if (validatedSchema.all) {
+    runBatchMigration(host, userLog);
+  } else {
+    runMigrationOnProject(host, validatedSchema, userLog);
+  }
+
+  formatFiles(host);
+
+  return () => {
+    printUserLogs(userLog);
+  };
+}
+
+function runMigrationOnProject(host: Tree, schema: ValidatedSchema, userLog: UserLog) {
+  const options = normalizeOptions(host, schema);
+  const packageJsonPath = options.paths.packageJson;
+  userLog.push({ type: 'info', message: `Convering ${options.name} v9 RC dependencies to carets` });
+
+  updateJson(host, packageJsonPath, (packageJson: PackageJson) => {
+    if (!packageJson.dependencies) {
+      userLog.push({ type: 'info', message: `${options.name} has no dependencies` });
+      return packageJson;
+    }
+
+    Object.entries(packageJson.dependencies).forEach(([dependency, dependencyRange]) => {
+      const range = semver.validRange(dependencyRange);
+      const minVersion = semver.minVersion(dependencyRange);
+
+      if (!minVersion) {
+        throw new Error(`${dependency} range ${dependencyRange} cannot be parsed to a min version`);
+      }
+
+      if (!isPackageVersionConverged(minVersion.raw) || minVersion.prerelease[0] !== 'rc' || range !== minVersion.raw) {
+        return packageJson;
+      }
+
+      packageJson.dependencies ??= {};
```

Comment:
Didn't know this groovy `??=` syntax was a thing until now 👀

Reply:
yeah it's far better than `variable = variable || {}` or `variable = variable ?? {}`